sdk/tests/deqp/framework/referencerenderer/rrFragmentOperations.js

```diff
@@ -196,7 +196,18 @@ rrFragmentOperations.executeDepthCompare = function(inputFragments, depthFunc, d
                 var fragSampleNdx = 0;
                 var depthBufferValue = depthBuffer.getPixDepth(fragSampleNdx, frag.pixelCoord[0], frag.pixelCoord[1]);
                 var sampleDepthFloat = frag.sampleDepths[fragSampleNdx];
-                var sampleDepth = deMath.clamp(sampleDepthFloat, 0.0, 1.0);
+
+                /* convert input float to target buffer format for comparison */
+                var access = new tcuTexture.PixelBufferAccess({
+                    format: depthBuffer.getFormat(),
+                    width: 1,
+                    height: 1,
+                    depth: 1,
+                    data: new ArrayBuffer(8)
+                });
```

Comment:
Could the allocation of the PixelBufferAccess object be hoisted outside this loop so allocations aren't performed per-pixel?